webapp/app/forms/steps/steuerlotse_step.py

```diff
@@ -116,38 +121,43 @@ def create_form(self, request, prefilled_data):
 
         return self.form(form_data, **prefilled_data)
 
-    def render(self):
+    def render(self, **kwargs):
         """
         Renders a Form step. Use the render_info to provide all the needed data for rendering.
         """
         return render_template(
             template_name_or_list=self.template,
             form=self.render_info.form,
-            render_info=self.render_info
+            render_info=self.render_info,
+            **kwargs
         )
 
     @staticmethod
-    def _delete_dependent_data(data_field_prefixes: list, stored_data: dict):
-        for field in list(stored_data.keys()):
-            if any([field.startswith(data_field_prefix) for data_field_prefix in data_field_prefixes]):
-                stored_data.pop(field)
-        return stored_data
+    def _delete_dependent_data(data_field_identifier: list, stored_data: dict):
+        """This method filters the stored data. It deletes all the elements where the key includes the
+        data_field_identifier. """
+        return dict(filter(
+            lambda elem: not any([elem[0].startswith(data_field_prefix) for data_field_prefix in data_field_identifier])
+                     and not any([elem[0].endswith(data_field_prefix) for data_field_prefix in data_field_identifier]),
+            stored_data.items()))
```

Comment:
> I could imagine that pseron_a_religion and steuerminderung_religion could become a problem)
Only if you are looking for religion, correct?

Oh, I had another look at your proposed changes and with them you would be correct. Actually, I think I would need to change the description: `This method filters the stored data. It deletes all the elements where the key begins or ends with the data_field_identifier.` and then maybe `data_fied_subfix`?


Nevertheless, we do not need to change this function because we now do delete the data differently for the eligibility steps.

Reply:
Add prefix + postfix parameters